### PR TITLE
plugins: set up auto-generated API docs

### PR DESCRIPTION
**Pull request by oandregal**
**Related to** https://github.com/WordPress/gutenberg/issues/14227

This PR sets up auto-generated API docs for the `plugins` package through the `npm run docs:build` command and updates its README file.

## Test instructions

- Delete or modify the contents at `packages/plugins/README.md` within the start `<!-- START TOKEN(Autogenerated API docs) -->` and end `<!-- END TOKEN(Autogenerated API docs) -->` tokens.
- Execute `npm run docs:build` and verify they are re-created.
- Review that the API docs match the JSDoc of the exported symbols.

### CODE REVIEW

**Comment by gziolo** [comment]
The downside of this change is that it no longer provides ES5 examples.

**Comment by nosolosw** [comment]
Yeah, we can provide multiple examples. I can bring those back. I thought that we wanted ES5/ES6 examples in the tutorials but we were fine with ES6-only in the npm packages.

**Comment by aduth** [comment]
> I thought that we wanted ES5/ES6 examples in the tutorials but we were fine with ES6-only in the npm packages.

This seems like it could be agreeable. Or at least, I have worries about (a) maintenance cost of keeping these examples in sync and (b) consistency of how we provide examples (i.e. all packages should either be consistently documented as both ES5/ES2015+ or as only ES2015+).

**Comment by gziolo** [comment]
> This seems like it could be agreeable. Or at least, I have worries about (a) maintenance cost of keeping these examples in sync and (b) consistency of how we provide examples (i.e. all packages should either be consistently documented as both ES5/ES2015+ or as only ES2015+).

If we pick ES2015+ + JSX, we should stop using WordPress globals in README files that are visible on npm and maybe find a way to replace them with `const { x } = wp.package;` syntax in Gutenberg handbook.

**Comment by nosolosw** [comment]
@gziolo @aduth To keep this simple: I've added the ES5 examples as well. This is how it'll look like:

screenshot from 2019-03-08 09-20-01

Note that there's no tabs. At least yet! I've found a way to generate them if we want to: it looks like we can add a `<caption>` to the [`@example`](http://usejsdoc.org/tags-example.html) that we can then useto do smart things like adding tabs. This logic needs to be added to the Markdown formatter https://github.com/WordPress/gutenberg/issues/14331

I don't think I can come back to this until later next week (the earliest) because I want to finish the zero-config task on time for next release. I propose to merge this PR as it is, and then iterate to bring back tabs if we want to.

File: packages/plugins/src/api/index.js [hunk 1]
```diff
@@ -77,6 +143,21 @@ export function registerPlugin( name, settings ) {
  *
  * @param {string} name Plugin name.
  *
+ * @example <caption>ES5</caption>
+ * ```js
+ * // Using ES5 syntax
```
Comment (gziolo) [inline packages/plugins/src/api/index.js]: It seems like `caption` and code comment duplicate each other. Is there an easy way to generate the comment or paragraph to simplify it?

Comment (nosolosw) [reply packages/plugins/src/api/index.js]: Yeah. As I mentioned, the idea will be that the caption is used for this, but it's not implemented at the moment.

**Review by gziolo — APPROVED** [review]
We can always iterate on the examples but the changes themselves for this particular README file look great 👍

File: packages/plugins/src/api/index.js [hunk 2]
```diff
@@ -77,6 +143,21 @@ export function registerPlugin( name, settings ) {
...
+ * var unregisterPlugin = wp.plugins.unregisterPlugin;
+ *
+ * unregisterPlugin( 'plugin-name' );
+ * ```
+ *
+ * @example <caption>ESNext</caption>
+ * ```js
+ * // Using ESNext syntax
+ * const { unregisterPlugin } = wp.plugins;
+ * unregisterPlugin( 'plugin-name' );
```
Comment (gziolo) [inline packages/plugins/src/api/index.js]: Nit: Missing an empty line to align with the other example.

Comment (nosolosw) [reply packages/plugins/src/api/index.js]: Fixed at 9d814ab and the automated process reminded me that I should also update the README. So I did at d15f919 Starting to :heart: it. Also: :sweat_smile:

**Pull request merged**